libvast/include/vast/pattern.hpp

```diff
@@ -48,7 +58,7 @@ class pattern {
 
   [[nodiscard]] const std::string& string() const;
 
-  [[nodiscard]] bool case_insensitive() const;
+  [[nodiscard]] const options& pattern_options() const;
```

Comment:
```suggestion
  [[nodiscard]] const pattern::options& options() const;
```

That would be nicer on the call site. (At the small cast of some qualifications within `pattern`.)

Reply:
Having an `options` struct with an `options()` method required me to add a `struct` qualifier for all `pattern::options` objects - I think it's much easier to keep `options()` but rename `pattern::options`  to `pattern::pattern_options` which I did in https://github.com/tenzir/vast/pull/2964/commits/fc832b5f8834326f11a94324db78ce868b53dd6f